test/grouping.jl

```diff
@@ -36,4 +36,6 @@ module TestGrouping
     x = pool(collect(1:20))
     df = DataFrame(v1=x, v2=x)
     groupby(df, [:v1, :v2])
+
+    by(e->1, DataFrame(x=[]), :x)
```

Comment:
Could you check the return value too? Could be good to test a function which actually uses the data (when present), like `sum`.